src/app/Caches/Create/ConfigurationFeature.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Divider,
+    Form,
+    FormGroup,
+    Grid,
+    GridItem,
+    InputGroup,
+    Text,
+    TextContent,
+    TextVariants,
+    TextInput,
+    Select,
+    SelectOption,
+    SelectVariant,
+} from '@patternfly/react-core';
+import { CacheFeature, MaxSizeUnit, EvictionStrategy } from "@services/infinispanRefData";
+import { useTranslation } from 'react-i18next';
+import { MoreInfoTooltip } from '@app/Common/MoreInfoTooltip';
+
+const ConfigurationFeature = (props: {
+    cacheFeature: CacheFeatureStep,
+    cacheFeatureModifier: (CacheFeatureStep) => void
+}) => {
+
+    const { t } = useTranslation();
+
+    const [cacheFeatureSelected, setCacheFeatureSelected] = useState(props.cacheFeature.cacheFeatureSelected);
+
+    //Bounded Cache
+    const [maxSize, setMaxSize] = useState<string | undefined>(props.cacheFeature.maxSize);
+    const [maxCount, setMaxCount] = useState<string | undefined>(props.cacheFeature.maxCount);
+    const [evictionStrategy, setEvictionStrategy] = useState<string>(props.cacheFeature.evictionStrategy);
+
+    const [isOpenCacheFeature, setIsOpenCacheFeature] = useState(false);
+    const [isOpenEvictionStrategy, setIsOpenEvictionStrategy] = useState(false);
+    const [isOpenMaxSizeUnit, setIsOpenMaxSizeUnit] = useState(false);
+
+    // Helper states for the maxSize input
+    const [maxSizeNumber, setMaxSizeNumber] = useState<string>();
+    const [maxSizeUnit, setMaxSizeUnit] = useState<MaxSizeUnit>(MaxSizeUnit.MB);
+
+    useEffect(() => {
+        if (maxSize) {
+            // Splitting the maxSize string into number and unit
+            const [num, unit] = maxSize.match(/\D+|\d+/g);
+            setMaxSizeNumber(num);
+            setMaxSizeUnit(MaxSizeUnit[unit]);
+        }
+    }, [])
+
+    useEffect(() => {
+        if (maxSizeNumber) {
+            setMaxSize(maxSizeNumber + maxSizeUnit);
+        }
+        if (maxSizeNumber === '') {
+            setMaxSize(undefined);
+            setMaxSizeNumber(undefined);
+        }
+        if (maxCount === '') setMaxCount(undefined);
+    }, [maxSizeNumber, maxSizeUnit, maxCount])
+
+    useEffect(() => {
+        props.cacheFeatureModifier({
+            cacheFeatureSelected: cacheFeatureSelected,
+            maxSize: maxSize,
```

Comment:
max size can be set to negatif number and in this case cache creation fails.
the max count gets disabled too